spec/validators_spec.rb

```diff
@@ -22,4 +22,26 @@ module Contracts
       end.to raise_error(ContractError, /Expected: 1\.\.10.*Actual: "hello world"/m)
     end
   end
+
+  describe "Regexp" do
+    it "should pass for a matching string" do
+      expect { o.should_contain_foo("containing foo") }.to_not raise_error
+    end
+
+    it "should fail for a non-matching string" do
+      expect { o.should_contain_foo("that's not F00") }.to raise_error(ContractError)
+    end
+
+    describe 'within a hash' do
+      it "should pass for a matching string" do
+        expect { o.hash_containing_foo(host: "foo.example.org") }.to_not raise_error
```

Comment:
Can you use here `1.8` hash rocket syntax, since we still want to support `1.8` ?